tests/ndarrays/leaks_check.py

```diff
@@ -0,0 +1,31 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+from numpy import array
+
+def create_array():
+    a = array([[1, 2, 3, 4, 5], [6, 7, 8, 9, 10]])# pylint: disable=unused-variable
+
+def array_to_pointer():
+    a = array([[1, 2, 3, 4, 5], [6, 7, 8, 9, 10]])
+    b = a# pylint: disable=unused-variable
+
+def view_assign():
+    a = array([[1, 2, 3, 4, 5], [6, 7, 8, 9, 10]])
+    b = a[1:]# pylint: disable=unused-variable
+
+def pointer_to_pointer():
+    a = array([[1, 2, 3, 4, 5], [6, 7, 8, 9, 10]])
+    b = a
+    c = b# pylint: disable=unused-variable
+
+def pointer_reassign():
+    a = array([[1, 2, 3, 4, 5], [6, 7, 8, 9, 10]])
+    c = array([[1, 2, 3, 4, 5, 0], [0, 6, 7, 8, 9, 10]])
+    b = a# pylint: disable=unused-variable
+    b = a[1:]
+    b = c
+
+create_array()
+array_to_pointer()
+view_assign()
+pointer_to_pointer()
+pointer_reassign()
```

Comment:
Maybe you could add an array created in a program instead of only in a function too?